libs/utils/utils.js

```diff
@@ -1150,6 +1150,16 @@ async function documentPostSectionLoading(config) {
   document.body.appendChild(createTag('div', { id: 'page-load-ok-milo', style: 'display: none;' }));
 }
 
+export function partition(arr, fn) {
+  return arr.reduce(
+    (acc, val, i, ar) => {
+      acc[fn(val, i, ar) ? 0 : 1].push(val);
+      return acc;
+    },
+    [[], []],
+  );
+}
```

Comment:
I know that this is probably the most efficient implementation of `partition`, but I think that considering the arrays being partitioned will always be small, I think readability should take precedence.  Also I think any testing of it can be part of general unit tests given it's simplicity.

cc @mokimo 

```suggestion
function partition(arr, fn) {
  return [
    arr.filter(fn),
    arr.filter((el, i) => !fn(el, i)),
  ];
}
```

Reply:
+1ing @chrischrischris opinion

Reply:
There could be a middle ground to make it more readable [similar to the haskell implementation](https://hackage.haskell.org/package/ghc-internal-9.1001.0/docs/src/GHC.Internal.Data.OldList.html#partition)

```javascript

function partition(arr, predicate) {
    const select = ([ts, fs], x, ...args) => {
      if (predicate(x, ...args)) {
        ts.push(x);
      } else {
        fs.push(x);
      }
      return [ts, fs];
    };
    
  return arr.reduce(select, [[], []]);
}
```

The `select` tells the reader that we're folding over the array and 'selecting' the values matching the predicate, so the really only need to read the last line to understand the function.